### PR TITLE
api: auto fill group id in rules when unset

### PR DESCRIPTION
**Pull request by duduainankai**
Signed-off-by: Zheming Li <nkdudu@126.com>

### What problem does this PR solve?

#3054 

### What is changed and how it works?

- Using `GroupBundle.ID` to auto fill `GroupId` field in Rule when unset.

### Check List

Tests

- Unit test

Code changes

- Has HTTP API interfaces change

### Release note

- No release note

### CODE REVIEW

File: server/api/rule.go [hunk 1]
```diff
@@ -454,6 +454,9 @@ func (h *ruleHandler) SetAllGroupBundles(w http.ResponseWriter, r *http.Request)
 				h.rd.JSON(w, http.StatusBadRequest, err.Error())
 				return
 			}
+			if len(rule.GroupID) != 0 {
```
Comment (disksing) [inline server/api/rule.go:457]: `!=` -> `==`?

**Comment by disksing** [comment]
I think we can auto set `bundle.ID` when it is empty to avoid this error:
```
	if group.ID != groupID {
		h.rd.JSON(w, http.StatusBadRequest, fmt.Sprintf("group id %s does not match request URI %s", group.ID, groupID))			h.rd.JSON(w, http.StatusBadRequest, fmt.Sprintf("group id %s does not match request URI %s", group.ID, groupID))
		return			return
	}		}
```

File: server/api/rule.go (hunk 1 above)
Comment (duduainankai) [reply server/api/rule.go:457]: Yep, a stupid mistake :(

**Comment by duduainankai** [comment]
> I think we can auto set `bundle.ID` when it is empty to avoid this error:
> 
> ```
> 	if group.ID != groupID {
> 		h.rd.JSON(w, http.StatusBadRequest, fmt.Sprintf("group id %s does not match request URI %s", group.ID, groupID))
> 	}
> ```

I have a second think here when wrote the code, maybe it's safer to keep this check ?

**Comment by disksing** [comment]
> > I think we can auto set `bundle.ID` when it is empty to avoid this error:
> > ```
> > 	if group.ID != groupID {
> > 		h.rd.JSON(w, http.StatusBadRequest, fmt.Sprintf("group id %s does not match request URI %s", group.ID, groupID))
> > 	}
> > ```
> 
> I have a second think here when wrote the code, maybe it's safer to keep this check ?

IMO it is the same with the changes we apply to `rule`. It should be ok if we only set it when it is empty.

**Comment by duduainankai** [comment]
> > > I think we can auto set `bundle.ID` when it is empty to avoid this error:
> > > ```
> > > 	if group.ID != groupID {
> > > 		h.rd.JSON(w, http.StatusBadRequest, fmt.Sprintf("group id %s does not match request URI %s", group.ID, groupID))
> > > 	}
> > > ```
> > 
> > 
> > I have a second think here when wrote the code, maybe it's safer to keep this check ?
> 
> IMO it is the same with the changes we apply to `rule`. It should be ok if we only set it when it is empty.

Done. PTAL.

**Review by disksing — APPROVED** [review]
Great! Thanks!

**Pull request merged**